### PR TITLE
UI-1149 DataGrid Theming

### PR DESCRIPTION
**Pull request by mallik-svmx**
- Applied SLDS Table/DataGrid Theming(sort Icon, title uppercase, coloring, layout etc.,)
- Adding configurable prop to following DataGrid/Table features:
-   - Column border
-   - Row border
-   - Zebra/striped rows
-   - Row hover
- Storybook examples with above configurable props
- Minor fix to Badge click outline css

### CODE REVIEW

File: lib/components/DataGrid/DataGrid.scss [hunk 1]
```diff
@@ -12,17 +24,32 @@
...
   }
+
+  .rt-th {
+    padding: 0.75rem 0;
+    background-color: #fafaf9;
+    text-transform: uppercase;
+  }
+}
+
+.DataGrid .rt-thead .rt-th.-sort-asc .rt-resizer, {
```
Comment (sheela-svmx) [inline lib/components/DataGrid/DataGrid.scss]:
``` .DataGrid .rt-thead {
  .rt-th {
      &.-sort-asc{
        .rt-resizer {
          background: no-repeat center/50% url('/assets/icons/utility/arrowdown_60.png');
        }
      }
      &.-sort-desc { 
        .rt-resizer {
          background: no-repeat center/50% url('/assets/icons/utility/arrowdown_60.png');
        }
      }
  }
}``` 

Minor but might be more readable when nested  plus we are clearly saying with &-sort-asc that we have 2 diff background for the two cases

File: lib/components/DataGrid/DataGrid.scss [hunk 2]
```diff
@@ -12,17 +24,32 @@
...
+  background: no-repeat center/50% url('/assets/icons/utility/arrowdown_60.png');
+}
+
+.DataGrid .rt-thead .rt-th.-sort-desc .rt-resizer, {
```
Comment (sheela-svmx) [inline lib/components/DataGrid/DataGrid.scss]:
``` 
 .rt-th {
      &.-sort-asc{
        .rt-resizer {
          background: no-repeat center/50% url('/assets/icons/utility/arrowdown_60.png');
        }
      }
      &.-sort-desc { 
        .rt-resizer {
          background: no-repeat center/50% url('/assets/icons/utility/arrowdown_60.png');
        }
      }
  }
}``` 

Minor but might be more readable when nested  plus we are clearly saying with &-sort-asc that we have 2 diff background for the two cases

**Comment by sheela-svmx** [comment]
@mallik-svmx my comment is minor but just makes it more readable for nested css imo

File: lib/components/DataGrid/DataGrid.scss (hunk 2 above)
Comment (mallik-svmx) [reply lib/components/DataGrid/DataGrid.scss]: 👍 Done.  Just so, we know, we shouldn't be going more than 3 level nesting, as best practice and due to some performance implications.

**Comment by mallik-svmx** [comment]
Merging with two approvals.

**Pull request merged**